[test] Adds clean-up for functional test 02167

tonickkozlov opened the pull request:
Running this test twice now doesn't result in a failure due to the fact that the temporary file is being cleaned up

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Clean-up after functional test 02167
...


> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

tonickkozlov committed:
[test] Adds clean-up for functional test 02167

Running this test twice now doesn't result in a failure due to the fact that the temporary file is being cleaned up

davenger commented on tests/queries/0_stateless/02167_format_from_file_extension.sh:
```diff
@@ -5,6 +5,12 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 
+function cleanup()
+{
+    user_files_path=$(clickhouse-client --query "select _path,_file from file('nonexist.txt', 'CSV', 'val1 char')" 2>&1 | grep Exception | awk '{gsub("/nonexist.txt","",$9); print $9}')
```
Also I'd suggest to add a comment that this command expects an error message like 
`Code: 107. DB::Exception: Received from localhost:9000. DB::Exception: File /home/davenger/src/ClickHouse/build_Debug/user_files/nonexist.txt doesn't exist. (FILE_DOESNT_EXIST)`
So that if this message changes in future it will be easier to fix this command
```suggestion
    user_files_path=$($CLICKHOUSE_CLIENT --query "select _path,_file from file('nonexist.txt', 'CSV', 'val1 char')" 2>&1 | grep FILE_DOESNT_EXIST | awk '{gsub("/nonexist.txt","",$9); print $9}')
```

tonickkozlov committed:
Updated a comment that clarifies the expected error

davenger commented:
@mergifyio update

mergify[bot] committed:
Merge branch 'master' into patch-2

davenger committed:
Fix user file path extraction

It turned out that $CLICHOUSE_CLIENT could produce more verbose output that was not properly parsed. I made a change that should fix this parsing.

davenger committed:
Removed debug print
